### PR TITLE
Print conflicting toolchain files

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

The package build scheduler will print a list of files that it re-built that conflict with the toolchain manifests. No RPMs listed in the toolchain manifests should rebuild. Building them is the sole responsibility of the toolchain scripts.

###### Change Log  

- Add `--reserved-file-list-file` which should be a list of *.rpm filenames. By default this is the toolchain manifest file for the current build.
 - The build report at the end of the build will now include  `RPM Conflicts with toolchain` and `SRPM Conflicts with toolchain` sections if any such conflicts exist.

###### Does this affect the toolchain?  

**NO**

### CODE REVIEW

File: toolkit/tools/scheduler/scheduler.go [hunk 1]
```diff
@@ -178,7 +188,7 @@ func cancelBuildsOnSignal(signals chan os.Signal, agent buildagents.BuildAgent)
 
 // buildGraph builds all packages in the dependency graph requested.
 // It will save the resulting graph to outputFile.
-func buildGraph(inputFile, outputFile string, agent buildagents.BuildAgent, workers, buildAttempts int, stopOnFailure, canUseCache bool, packagesToBuild []*pkgjson.PackageVer, packagesNamesToRebuild, ignoredPackages []string) (err error) {
+func buildGraph(inputFile, outputFile string, agent buildagents.BuildAgent, workers, buildAttempts int, stopOnFailure, canUseCache bool, packagesToBuild []*pkgjson.PackageVer, packagesNamesToRebuild, ignoredPackages, reservedFiles []string) (err error) {
```
Comment (anphel31) [inline toolkit/tools/scheduler/scheduler.go:191]: should ignorgedPackages still be listed with "[]string" ?

File: toolkit/tools/scheduler/schedulerutils/printresults.go [hunk 2]
```diff
@@ -48,6 +48,8 @@ func PrintBuildSummary(pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMutex, bu
 	builtSRPMs := make(map[string]bool)
```
Comment (anphel31) [inline toolkit/tools/scheduler/schedulerutils/printresults.go:48]: General question for this function - can we log the outputs to a file? It seems like a useful artifact to publish along with the other logs

**Review by anphel31 — APPROVED** [review]

File: toolkit/tools/scheduler/schedulerutils/printresults.go (hunk 2 above)
Comment (dmcilvaney) [reply toolkit/tools/scheduler/schedulerutils/printresults.go:48]: I'll add to my todo list

File: toolkit/tools/scheduler/scheduler.go (hunk 1 above)
Comment (dmcilvaney) [reply toolkit/tools/scheduler/scheduler.go:191]: Both are the same type (`[]string`). I think this is fine.

**Pull request merged**